Doc/whatsnew/3.11.rst

```diff
@@ -126,6 +126,10 @@ Deprecated
 Removed
 =======
 
+* The :func:`asyncio.coroutine` :term:`decorator` enabling legacy
+  generator-based coroutines to be compatible with async/await code.
+  Use :keyword:`async def` instead.
```

Comment:
Please mention that it's deprecated since Python 3.8 and the removal was initialially scheduled in Python 3.10.

Update also the NEWS entry.